engine/src/main/java/com/datatorrent/stram/StramLocalCluster.java

```diff
@@ -308,6 +313,24 @@ public StramLocalCluster(LogicalPlan dag) throws IOException, ClassNotFoundExcep
     this.dnmgr = new StreamingContainerManager(dag);
     this.umbilical = new UmbilicalProtocolLocalImpl();
 
+    Method addURL;
+    URLClassLoader loader;
+    try {
+      loader = (URLClassLoader)ClassLoader.getSystemClassLoader();
+      addURL = URLClassLoader.class.getDeclaredMethod("addURL", new Class[] { URL.class });
+      addURL.setAccessible(true);
+    } catch (NoSuchMethodException e) {
+      throw new RuntimeException("No method named addURL found.", e);
+    }
+
+    for (String jarPath : dag.getJarResources().keySet()) {
+      try {
+        addToClasspath(jarPath, loader, addURL);
+      } catch (InvocationTargetException | IllegalAccessException e) {
```

Comment:
Actually I take my word back. There are few tests which do not handle "Exception".

These are some to name a few.
engine/src/test/java/com/datatorrent/stram/CheckpointTest.java
engine/src/test/java/com/datatorrent/stram/engine/StreamingContainerTest.java
engine/src/test/java/com/datatorrent/stram/plan/StreamPersistanceTests.java

Please let me know if its OK to modify these tests as a part of this pull request to catch Exceoption. I'll make the changes. Till then I'll throw RuntimeException.